internal/pkg/cli/deploy/workload.go

```diff
@@ -355,6 +368,52 @@ func (img ContainerImageIdentifier) Tag() string {
 	return img.GitShortCommitTag
 }
 
+// buildPushOutputBuffer contains the output configuration for docker build and push.
+type buildPushOutputBuffer struct {
+	ContainerName string // ContainerName is the name of the container.
+
+	bufMu sync.Mutex // bufMu is a mutex to protect access to the buffer.
+
+	buf bytes.Buffer // buf is the buffer containing the output of build and push.
+
+	doneMu sync.Mutex // doneMu is a mutex to protect access to the done field.
+
+	done bool // done is a flag indicating whether the build and push is completed.
+}
+
+// Write appends the given bytes to the buffer.
+func (b *buildPushOutputBuffer) Write(p []byte) (n int, err error) {
+	b.bufMu.Lock()
+	defer b.bufMu.Unlock()
+	return b.buf.Write(p)
+}
+
+// logs returns the label (i.e., the first line of the buffer) and the last five lines of the buffer.
+func (b *buildPushOutputBuffer) logs() (string, [maxLogLines]string) {
+	b.bufMu.Lock()
+	defer b.bufMu.Unlock()
+
+	// Split the buffer bytes into lines
+	lines := strings.Split(strings.TrimSpace(b.buf.String()), "\n")
+
+	// Determine the start and end index to extract last 5 lines
+	start := 1
+	if len(lines) > maxLogLines {
+		start = len(lines) - maxLogLines
+	}
+	end := len(lines)
+
+	// Extract the last 5 lines and store them in a fixed-size array
+	var logLines [maxLogLines]string
+	idx := 0
+	for start < end {
+		logLines[idx] = strings.TrimSpace(lines[start])
+		start++
+		idx++
+	}
+	return lines[0], logLines
```

Comment:
if `len(lines) < maxLogLines`, won't `logLines` contain `lines[0]`? i wonder if the label line should just be a fixed string on `buildPushOutputBuffer`?

Reply:
ah i see why it's not a fixed string, though the double-printing issue still stands 👍